src/android/SonectCordovaPlugin.java

```diff
@@ -3,26 +3,69 @@
 import org.apache.cordova.CallbackContext;
 import org.apache.cordova.CordovaArgs;
 import org.apache.cordova.CordovaPlugin;
+import org.json.JSONArray;
 import org.json.JSONException;
+import org.json.JSONObject;
 
 import java.util.ArrayList;
 
 public class SonectCordovaPlugin extends CordovaPlugin {
+
+    private static boolean IS_LIGHT_THEME;
+    private static ArrayList<SDKEntryPointActivity.PaymentMethodReference> PAYMENT_METHODS;
+    private static SonectSDK.Config.UserCredentials CREDENTIALS;
+
     @Override
     public boolean execute(String action, CordovaArgs args, CallbackContext callbackContext) throws JSONException {
-        if ("present".equalsIgnoreCase(action)) {
-            ArrayList<SDKEntryPointActivity.PaymentMethodReference> pms = new ArrayList<>(1);
-            // TODO define JS interface with NEON and then provide valid payment method
-            pms.add(new SDKEntryPointActivity.PaymentMethodReference(
-                            "My payment method", R.drawable.ic_visa, "**** 1234", 302.44f
+        if (!("init".equalsIgnoreCase(action)) && CREDENTIALS == null) {
+            throw new IllegalStateException("Call init() first");
+        }
+        if ("init".equalsIgnoreCase(action)) {
+            processInitCommand(args);
+        } else if ("present".equalsIgnoreCase(action)) {
+            processPresentCommand(args);
+        } else if ("presentTransaction".equalsIgnoreCase(action)) {
+            processPresentTransactionCommand(args);
+        }
+        return true;
+    }
+
+    private void processInitCommand(CordovaArgs args) throws JSONException {
+        JSONObject credentials = args.getJSONObject(0);
+        JSONArray paymentMethods = args.getJSONArray(1);
+        JSONObject theme = args.getJSONObject(2);
+
+        PAYMENT_METHODS = new ArrayList<>();
+        for (int i = 0; i < paymentMethods.length(); i++) {
+            JSONObject pm = paymentMethods.getJSONObject(i);
+            PAYMENT_METHODS.add(new SDKEntryPointActivity.PaymentMethodReference(
+                            pm.getString("name"), R.drawable.ic_sonect_web_bank, pm.getString("detailDescription"), (float) pm.getDouble("funds")
                     )
             );
-            SDKEntryPointActivity.Companion.start(cordova.getActivity(),
-                    new SonectSDK.Config.UserCredentials(args.getString(1),
-                            args.getString(0), args.getString(2), null), pms);
         }
-        return true;
+
+        CREDENTIALS = new SonectSDK.Config.UserCredentials(credentials.getString("userId"),
+                credentials.getString("token"),
+                credentials.getString("signature"), "eu", null);
+
+        IS_LIGHT_THEME = theme.get("type") == "light";
```

Comment:
As of now we're only using dark/light switchers without details I guess. Will adjust it later on, I guess we could control appbar colors as well.